dnf/base.py

```diff
@@ -665,6 +669,7 @@ def do_transaction(self, display=()):
 
         if not self.transaction:
             if self._group_persistor:
+                self._trans_success = True
```

Comment:
It means transaction here, but in my case I use it in case that there is no transaction but some groups are going to be marked as removed (packages has to be kept as requirement of other packages).